### PR TITLE
feat(react-infobutton, react-components): Release InfoLabel to stable and deprecate InfoButton in favor of InfoLabel

### LINKED ISSUE #25062: InfoLabel
## Preparation:

- [x] Open UI Research
  - N/A
- [x] Open GitHub issues related to component
  - N/A
- [x] react-\* package scaffolded with the right ownership in CODEOWNERS
  - #25083 
- [x] Component Spec authored and reviewed
  - #25251 

## Implementation

- [x] Component implementation
  - #25247  
- [x] Initial conformance and unit tests (validate basic functionality)
- [x] Initial documentation
  - [x] README.md covering basic usage
    - #25686 
  - [x] MIGRATION.md guide (include v8 and v0)
    - N/A 
  - [x] Storybook stories
    - #25247 
- [x] Component released as preview
  - #29562

## Validation

- [x] Add tests
  - [x] Bundle size fixtures
    - #25083 
  - [x] Unit tests
    - #25247 
  - [x] Conformance tests
    - #25247 
    - #25682
  - [x] VR tests
    - #26305
  - [x] Performance test scenario
    - #26271 
  - [ ] Accessibility behavior tests
  - [x] Create an issue and run [manual accessibility tests](https://github.com/microsoft/fluentui/wiki/Manual-Accessibility-Review-Checklist): 
    - #26427 
- [x] Validate with partners
- [x] Run a bug bash with other FUI crews
- [x] Finalize documentation
  - [x] Review and add any missing storybook
  - [x] Finzalize migration guide (N/A)
- [x] Component released as stable
  - [x] Run the `prepare-initial-release` script to remove the `-preview` suffix from the package name  #29585
  - [ ] Deprecate the `-preview` package in NPM
  - [x] Export the component from `@fluentui/react-components`  #29585
  - [x] In package.json: Remove the alpha/beta tag from the version number in package.json  #29585
  - [x] In package.json: Change beachball's disallowedChangeTypes to "major", "prerelease" #29585

### LINKED ISSUE #28615: InfoLabel: Move to stable
As part of this move, we will do the following:
- [x] #29479
- [x] #29544
- [x] #29557
- [x] #29561
- [x] #29584
- [x] #29602
- [x] #29604

### PR DESCRIPTION
**Pull request by sopranopillow**
## Changes

This PR Releases InfoLabel to stable and deprecates InfoButton to ensure use of the correctly handled a11y from InfoLabel.

## Related Issue(s)

- Fixes #28615 
- Fixes #25062

### CODE REVIEW

File: packages/react-components/react-infobutton/src/components/InfoButton/InfoButton.tsx [hunk 1]
```diff
@@ -7,6 +7,8 @@ import type { InfoButtonProps } from './InfoButton.types';
 
 /**
  * InfoButtons provide a way to display additional information about a form field or an area in the UI.
+ *
+ * @deprecated Use `InfoLabel` instead.
```
[the remark this answers is not in the document]
Comment (sopranopillow) [reply packages/react-components/react-infobutton/src/components/InfoButton/InfoButton.tsx:11]: I am open to ideas about how to handle this, but I do think deprecating is the way to go to have an accessible out of the box component.

File: apps/vr-tests-react-components/src/stories/InfoButton.stories.tsx
Comment (sopranopillow) [inline apps/vr-tests-react-components/src/stories/InfoButton.stories.tsx:1]: Note removing these tests due to InfoLabel already being tested and InfoButton will only be used inside InfoLabel

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Comment by sopranopillow** [comment]
Note, there's a bug in Popover that currently affects InfoLabel #29185. This needs to be fixed before merging.

**Review by AtishayMsft — DISMISSED** [review]
Approved on behalf of react charting team

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 2]
```diff
@@ -533,7 +533,7 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
       this._classNames = getClassNames(this.props.styles!, {
         theme: this.props.theme!,
         legendColor: this.state.color,
-        shouldHighlight: shouldHighlight,
```
Comment (Hotell) [inline packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx]: can we undo those react-charting lint autofix changes ? they are not related and are confusing for this particular PR which will release stable v9 controll

File: packages/react-experiments/src/components/Pagination/Pagination.base.tsx [hunk 3]
```diff
@@ -58,7 +58,7 @@ export class PaginationBase extends React.Component<IPaginationProps> {
 
     this._classNames = getClassNames(styles!, {
       theme: theme!,
-      format: format,
```
Comment (Hotell) [inline packages/react-experiments/src/components/Pagination/Pagination.base.tsx]: same as react-charting comment

File: packages/react-components/react-infobutton/src/components/InfoButton/InfoButton.tsx (hunk 1 above)
Comment (Hotell) [inline packages/react-components/react-infobutton/src/components/InfoButton/InfoButton.tsx:11]: can we use `@lint` jsdoc for better DX ? so when one clicks on this message they will be redirected to InfoLabel

File: packages/react-components/react-infobutton/README.md [hunk 4]
```diff
@@ -1,29 +1,25 @@
 # @fluentui/react-infobutton
 
-**React Infobutton components for [Fluent UI React](https://react.fluentui.dev/)**
-
-## STATUS: WIP 🚧
-
-These are not production-ready components and **should never be used in product**. This space is useful for testing new components whose APIs might change before final release.
+**React InfoLabel components for [Fluent UI React](https://react.fluentui.dev/)**
```
Comment (Hotell) [inline packages/react-components/react-infobutton/README.md:3]: hmm maybe we should create a new package that will include InfoLabel only ? 

with this approach this is a bit confusing. react-infobutton package exposes only `InfoLabel` , why is it called infobutton then ?

wdyt ?

Comment (sopranopillow) [reply packages/react-components/react-infobutton/README.md:3]: I think that's fair, the only thing is that this package was created before the unstable exports issues. So InfoLabel and InfoButton will still be under unstable and we would have to deprecate both. I can definitely make the `react-infolabel` package, but with those concerns, what are your thoughts?

Comment (sopranopillow) [reply packages/react-components/react-infobutton/README.md:3]: After seeing the reply to my comment above, I see why this makes sense. I do want to know what your thoughts on basically having an unusable react-infobutton in our repo and react-components? We would create this second package, but react-infobutton would stay there with no purpose, except to not cause a breaking change.

**Comment by sopranopillow** [comment]
After discussing offline with Martin, we've decided to create a react-infolabel package and deprecate react-infobutton to better align with the content we want to expose. Thanks for bringing this up Martin :)

**Pull request closed without merging**